lld/MachO/InputFiles.cpp

```diff
@@ -1522,13 +1522,17 @@ void ObjFile::registerEhFrames(Section &ehFrameSection) {
 }
 
 std::string ObjFile::sourceFile() const {
+  const char *unitName = compileUnit->getUnitDIE().getShortName();
```

Comment:
I would have wondered why the logic was written this way if your commit hadn't mentioned that it was matching the check in libObject :) Could you add a code comment here pointing to that logic so that it's easy to understand why it's written this way?

Reply:
Seems more robust just to use the same comment as there.

Reply:
Sure, works for me.